lib/puppet/network/http/compression.rb

```diff
@@ -20,7 +20,8 @@ module Active
     def uncompress_body(response)
       case response['content-encoding']
       when 'gzip'
-        return Zlib::GzipReader.new(StringIO.new(response.body)).read
+        # ZLib::GzipReader has an associated encoding, by default Encoding.default_external
+        return Zlib::GzipReader.new(StringIO.new(response.body), :encoding => Encoding::BINARY).read
```

Comment:
Actually, for `StringIO.new`, it doesn't appear that specifying an encoding in the constructor doesn't do anything? It appears it will just take on the encoding of the string (first argument)?:
```ruby
[26] pry(main)> binary = "foo".force_encoding(Encoding::BINARY)
=> "foo"
[27] pry(main)> io = StringIO.new(binary, 'rb')
=> #<StringIO:0x007f9bfbadb800>
[28] pry(main)> io.external_encoding
=> #<Encoding:ASCII-8BIT>
[29] pry(main)> read = io.read
=> "foo"
[30] pry(main)> read.encoding
=> #<Encoding:ASCII-8BIT>
[31] pry(main)> utf8 = "foo".force_encoding(Encoding::UTF_8)
=> "foo"
[32] pry(main)> io = StringIO.new(utf8, 'rb')
=> #<StringIO:0x007f9bfb9985b0>
[33] pry(main)> io.external_encoding
=> #<Encoding:UTF-8>
[34] pry(main)> read = io.read
=> "foo"
[35] pry(main)> read.encoding
=> #<Encoding:UTF-8>
```